thingsboard_gateway/tb_utility/tb_gateway_remote_configurator.py

```diff
@@ -60,6 +60,12 @@ def process_configuration(self, configuration):
                 self.__new_configuration = loads(decoded_configuration)
                 self.__old_connectors_configs = self.__gateway.connectors_configs
                 self.__new_general_configuration_file = self.__new_configuration.get("thingsboard")
+                
+                # To maintain RemoteShell status
+                old_general_configuration = loads(b64decode(self.__old_configuration))
```

Comment:
The added line is in line 65, before `self.__old_configuration` is assigned to `self.__new_configuration`
I tried to print out `self.__old_configuration` before line 65 and it seems it was base64 encoded.
Here I was trying to check if 'remoteShell' was defined in the old configuration file, if yes, then the new config should take the same value as the old config. Otherwise, the 'remoteShell' will always be removed in new config and remote shell function will be lost.
Or is there another better way to check if 'remoteShell' was defined in the old configuration file?

Reply:
Yes, I have checked, you are right, than please change assignment from `self.__old_configuration = encoded_current_configuration` in method send_current_configuration to `self.__old_configuration = json_current_configuration`